kvdb/etcd/readwrite_bucket_test.go

```diff
@@ -405,76 +405,6 @@ func TestBucketSequence(t *testing.T) {
 	require.Nil(t, err)
 }
 
-// TestKeyClash tests that one cannot create a bucket if a value with the same
-// key exists and the same is true in reverse: that a value cannot be put if
-// a bucket with the same key exists.
-func TestKeyClash(t *testing.T) {
-	t.Parallel()
-
-	f := NewEtcdTestFixture(t)
-	defer f.Cleanup()
-
-	db, err := newEtcdBackend(context.TODO(), f.BackendConfig())
-	require.NoError(t, err)
-
-	// First:
-	// put: /apple/key -> val
-	// create bucket: /apple/banana
-	err = db.Update(func(tx walletdb.ReadWriteTx) error {
-		apple, err := tx.CreateTopLevelBucket([]byte("apple"))
-		require.Nil(t, err)
-		require.NotNil(t, apple)
-
-		require.NoError(t, apple.Put([]byte("key"), []byte("val")))
-
-		banana, err := apple.CreateBucket([]byte("banana"))
-		require.Nil(t, err)
-		require.NotNil(t, banana)
-
-		return nil
-	}, func() {})
-
-	require.Nil(t, err)
-
-	// Next try to:
-	// put: /apple/banana -> val => will fail (as /apple/banana is a bucket)
-	// create bucket: /apple/key => will fail (as /apple/key is a value)
-	err = db.Update(func(tx walletdb.ReadWriteTx) error {
-		apple, err := tx.CreateTopLevelBucket([]byte("apple"))
-		require.Nil(t, err)
-		require.NotNil(t, apple)
-
-		require.Error(t,
-			walletdb.ErrIncompatibleValue,
-			apple.Put([]byte("banana"), []byte("val")),
-		)
-
-		b, err := apple.CreateBucket([]byte("key"))
-		require.Nil(t, b)
-		require.Error(t, walletdb.ErrIncompatibleValue, b)
-
-		b, err = apple.CreateBucketIfNotExists([]byte("key"))
-		require.Nil(t, b)
-		require.Error(t, walletdb.ErrIncompatibleValue, b)
```

Comment:
Can't these tests stay and assert what we do expect to happen?

Reply:
Discussed offline, put the checks back behind the `dev` build tag.